### PR TITLE
Make Examples workflows more granular

### PR DESCRIPTION
**Pull request by rwalker-apple**
#### Problem
 Examples jobs can't be rerun except as a monolithic "re-run all jobs", which takes a long time to retry in case of outages.

 #### Summary of Changes
 decompose examples into individual yamls

### CODE REVIEW

**Review by mspang — APPROVED** [review]
Should there be some punctuation here?

Build example (EFR32 Lock App for BRD4161A)

Build example: EFR32 Lock App for BRD4161A

Build example - EFR32 Lock App for BRD4161A

?

**Comment by rwalker-apple** [comment]
> Should there be some punctuation here?
> 
> Build example (EFR32 Lock App for BRD4161A)
> 
> Build example: EFR32 Lock App for BRD4161A
> 
> Build example - EFR32 Lock App for BRD4161A
> 
> ?

sure, will take a run at it

**Review by andy31415 — APPROVED** [review]

File: .github/workflows/examples-esp32.yaml [hunk 1]
```diff
@@ -12,13 +12,14 @@
...
 
-name: Examples - GN
+name: Build example - ESP32
 
 on:
     push:
     pull_request:
 
 jobs:
+    # TODO ESP32 https://github.com/project-chip/connectedhomeip/issues/1510
```
Comment (bzbarsky-apple) [inline .github/workflows/examples-esp32.yaml:22]: That issue is resolved; we build with gn.  There's no todo left here.

File: .github/workflows/examples-nrf.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            - name: Uploading Binaries
+              uses: actions/upload-artifact@v2
+              with:
+                  name:
+                      ${{ env.BUILD_TYPE }}-example-build-${{
+                      steps.outsuffix.outputs.value }}
+                  path: |
+                      out/lock_app_debug/chip-nrf52840-lock-example
+                      out/lighting_app_debug/chip-nrf52840-lighting-example
+            # TODO: NRF Connect https://github.com/project-chip/connectedhomeip/issues/2225
```
Comment (bzbarsky-apple) [inline .github/workflows/examples-nrf.yaml:76]: Copy paste, but this is dealt with now too.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by rwalker-apple** [comment]
@hawk248 @jelderton @saurabhst @BroderickCarlin ?

**Comment by rwalker-apple** [comment]
@hawk248 @jelderton @saurabhst @BroderickCarlin ?

**Comment by rwalker-apple** [comment]
@hawk248 @jelderton @saurabhst @BroderickCarlin ?

**Review by BroderickCarlin — APPROVED** [review]

**Comment by BroderickCarlin** [comment]
@rwalker-apple, once the merge conflicts are resolved it should be good

**Pull request merged**